core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -445,4 +458,12 @@ private UpdateResultHolder executeUpdateRequest(UpdateRequest updateRequest, Ind
                 primaryResponse.getSeqNo(), request.primaryTerm(), version, versionType);
         return replica.delete(delete);
     }
+
+    private Engine.NoOpResult executeNoOpRequestOnReplica(BulkItemResponse.Failure primaryFailure, DocWriteRequest docWriteRequest, IndexShard replica) throws IOException {
```

Comment:
I'm wondering if we should call this executeFailedSeqNoOnReplica .. might be a clearer name at this level?